Sources/Vapor/HTTP/Headers/HTTPHeaders+ContentRange.swift

```diff
@@ -0,0 +1,232 @@
+    case bytes
+    case custom(value: String)
+    
+    public func serialize() -> String {
+        switch self {
+            case .bytes:
+                return "bytes"
+            case .custom(let value):
+                return value
+        }
+    }
+}
+
+public struct RangeHeader : Equatable {
+    public let unit: RangeUnit
+    public let ranges: [RequestedRange]
+    
+    public init(unit: RangeUnit, ranges: [RequestedRange]) {
+        self.unit = unit
+        self.ranges = ranges
+    }
+    
+    public init?(requestHeader header: String) {
+        let splitResult = header.split(separator: "=", omittingEmptySubsequences: false)
+        guard splitResult.count == 2 else {
+            return nil
+        }
+        let (unitStr, rangesStr) = (splitResult[0], splitResult[1])
+        let ranges = rangesStr.split(separator: ",", omittingEmptySubsequences: false)
+            .map { it in RequestedRange.from(requestStr: it) }
+        if !ranges.filter({ $0 == nil }).isEmpty {
+            return nil
+        }
+        let lowerCasedUnit = unitStr.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
+        self.unit = lowerCasedUnit == "bytes"
+            ? RangeUnit.bytes
+            : RangeUnit.custom(value: lowerCasedUnit)
+        self.ranges = ranges.compactMap { $0 }
+    }
+    
+    func serialize() -> String {
+        return "\(unit.serialize())=\(ranges.map { $0.serialize() }.joined(separator: ", "))"
+    }
+    
+}
+
+public struct ContentRangeHeader : Equatable {
+    public let unit: RangeUnit
+    public let range: ResponseContentRange
+    
+    public init(unit: RangeUnit, range: ResponseContentRange) {
+        self.unit = unit
+        self.range = range
+    }
+    
+    public init?(responseHeader header: String) {
+        let splitResult = header.split(separator: " ", omittingEmptySubsequences: false)
+        guard splitResult.count == 2 else {
+            return nil
+        }
+        let (unitStr, rangeStr) = (splitResult[0], splitResult[1])
+        let lowerCasedUnit = unitStr.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
+        guard let contentRange = ResponseContentRange.from(responseStr: rangeStr) else {
+            return nil
+        }
+        self.unit = lowerCasedUnit == "bytes"
+            ? RangeUnit.bytes
+            : RangeUnit.custom(value: lowerCasedUnit)
+        self.range = contentRange
+    }
+    
+    func serialize() -> String {
+        return "\(unit.serialize()) \(range.serialize())"
+    }
+}
+
+public enum ResponseContentRange : Equatable {
+    case within(start: Int, end: Int)
+    case withinWithLimit(start: Int, end: Int, limit: Int)
+    case any(size: Int)
+    
+    public static func from<T>(responseStr: T) -> ResponseContentRange? where T : StringProtocol {
+        let ranges = responseStr.split(separator: "-", omittingEmptySubsequences: false)
+            .map({ $0.trimmingCharacters(in: .whitespacesAndNewlines) })
```

Comment:
Don't need the `()` around the closure here. 
```suggestion
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
```